camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java

```diff
@@ -73,17 +74,27 @@
     // stateful information
     private final AtomicInteger failures = new AtomicInteger();
     private final AtomicInteger state = new AtomicInteger(STATE_CLOSED);
+    private AtomicBoolean keepOpen = new AtomicBoolean(false);
     private volatile Timer halfOpenTimer;
     private volatile long lastFailure;
     private volatile long openedAt;
-    
+
     public ThrottlingExceptionRoutePolicy(int threshold, long failureWindow, long halfOpenAfter, List<Class<?>> handledExceptions) {
         this.throttledExceptions = handledExceptions;
         this.failureWindow = failureWindow;
         this.halfOpenAfter = halfOpenAfter;
         this.failureThreshold = threshold;
+        this.keepOpen.set(false);
```

Comment:
I would replace those statements above with `this(handledExceptions,failureWindow,halfOpenAfter,threshold,false)`.